### PR TITLE
feat: Current role data source

### PR DESCRIPTION
**Pull request by vishalvennu**
Added current_role data source.

## Test Plan

I had complied a provider and using Terraform was able to pull the current_role and id of a role I was using in Snowflake 
* [ ] acceptance tests

* [ ] …

### CODE REVIEW

File: pkg/datasources/current_role.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	"database/sql"
+	"fmt"
+	"log"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/snowflake"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var currentRoleSchema = map[string]*schema.Schema{
+	"current_role": {
```
Comment (sfc-gh-swinkler) [inline pkg/datasources/current_role.go:13]: can this be changed to "name"?

**Review by sfc-gh-swinkler — CHANGES_REQUESTED** [review]
looks good, but can the output be changed to "name" and a description added:

```
The name of the [primary role](https://docs.snowflake.com/en/user-guide/security-access-control-overview.html#label-access-control-role-enforcement) in use for the current session.
```

File: pkg/datasources/current_role.go (hunk 1 above)
Comment (vishalvennu) [reply pkg/datasources/current_role.go]: Done

File: docs/data-sources/current_role.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+## Example Usage
+
+```terraform
+data "snowflake_current_role" "this" {}
+
+resource "aws_ssm_parameter" "snowflake_account_role" {
+  name  = "/snowflake/account_role"
+  type  = "String"
+  value = data.snowflake_current_account.this.current_role
```
Comment (sfc-gh-swinkler) [inline docs/data-sources/current_role.md:20]: should be udpated to `data.snowflake_current_account.this.name`

**Review by sfc-gh-swinkler — CHANGES_REQUESTED** [review]

File: docs/data-sources/current_role.md (hunk 2 above)
Comment (vishalvennu) [reply docs/data-sources/current_role.md:20]: Yess, on it!

Comment (vishalvennu) [reply docs/data-sources/current_role.md]: Done

**Comment by vishalvennu** [comment]
Hello @sfc-gh-swinkler, the integration tests have failed because of a syntax error that got corrected in the latest commit. Could you please trigger the integration testing again? Thank you

**Comment by sfc-gh-swinkler** [comment]
im just going to go ahead and merge this and fix the go formatting issues. @vishalvennu thanks for your contribution!

**Comment by vishalvennu** [comment]
Thanks a bunch @sfc-gh-swinkler

**Pull request merged**